src/transformStateWithClones.js

```diff
@@ -6,8 +6,43 @@
  *
  * @return {Object[]}
  */
+
+const addProperties = (state, properties, res) => {
+  Object.assign(state, properties);
+  res.push({ ...state });
+};
+
+const removeProperty = (state, keysToRemove, res = []) => {
+  keysToRemove.forEach((key) => {
+    delete state[key];
+  });
+  res.push({ ...state });
+};
+
+const clearState = (state, res) => {
+  Object.keys(state).forEach((key) => delete state[key]);
+
+  res.push({ ...state });
+};
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const res = [];
```

Comment:
```suggestion
  const statesHistory = [];
```
`res/result/arr/obj/el/elem/` are bad names, variable names should be descriptive